be/dtm0_log.h

```diff
@@ -0,0 +1,236 @@
+/* -*- C -*- */
+/*
+ * Copyright (c) 2013-2020 Seagate Technology LLC and/or its Affiliates
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * For any questions about this software or licensing,
+ * please email opensource@seagate.com or cortx-questions@seagate.com.
+ *
+ */
+
+#pragma once
+#ifndef __MOTR_BE_DTM0_LOG_H__
+#define __MOTR_BE_DTM0_LOG_H__
+
+/**
+ *  @page dtm0 log implementation
+ *
+ *  @section Overview
+ *  DTM0 log module will be working on incoming message request, the
+ *  goal of this module is to track journaling of incoming message either on
+ *  persistent or volatile memory based on whether logging is happening on
+ *  participant or on originator, so that in the phase failure consistency of
+ *  failed participant can be restored by iterating over logged journal to
+ *  decide which of the logged messages needs to be sent as a redo request.
+ *
+ *  During normal operation when every participant and originator is online
+ *  every transaction will be logged and at any point state of transaction can
+ *  be any of the following,
```

Comment:
Suggestion: let's rewrite this paragraph (or its parts) in the following way:

A transaction has a group of states of each individual participant of the transaction (note: originator is not a part of this group). Each entry of such a group may have many different states, but this set of states must include the following states:
* InProgress - the node (owner of the log) has a request.
* Executed - the node has a request and the corresponding reply.
* Persistent - the node got a Persistent notice.

Rationale: 
1. A transaction does not have a dedicated state machine. Its state is inferred from the states of the participants. A participant have a dedicated state machine and transitions.
2. Unknown state cannot exist: the group of PA states is a part of any request/reply/REDO/PersistentNotice. A transaction is sent with PA states set to InProgress (at least) on the sender side.

Reply:
Addressed at 4e0d3688b2cc72748d9c1ed8df1ddeb8a2319fba